src/network/client.rs

```diff
@@ -54,19 +55,23 @@ where
         let mut peer_comms = PeerComms::new();
         let block_req = service.block_subscription(peer_comms.subscribe_to_block_announcements());
         let gossip_req = service.gossip_subscription(peer_comms.subscribe_to_gossip());
+        let err_logger = global_state.logger().clone();
```

Comment:
This and the instance in the `and_then` closure below could be augmented with the connection address. At the end the logger that goes inside the `Client` would further clone one of these instances to add the node ID, so we'll get both the remote address and the node ID in the context of the client logger.

Reply:
Good point. I've solved it slightly differently, instead of accepting `GlobalState` now it accepts `ConnectionState`, which has logger with address already there.